runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/binding/TestBindingFactory.java

```diff
@@ -188,11 +189,32 @@ public MessageConsumer newStream(
         long routedId = begin.routedId();
         long initialId = begin.streamId();
         long replyId = initialId ^ 1L;
-        long resolvedId = router.get(routedId);
 
         MessageConsumer newStream =  null;
+        long resolvedId;
 
-        if (resolvedId != -1L)
+        BindingConfig binding = bindings.get(routedId);
+
+        if (guard != null)
+        {
+            final long authorization = guard.reauthorize(begin.traceId(), routedId, 0, credentials);
+            List<TestBindingRouteConfig> routes = binding.routes.stream().map(TestBindingRouteConfig::new).collect(toList());
+            TestBindingRouteConfig route = routes.stream()
+                .filter(r -> r.authorized(authorization))
+                .findFirst()
+                .orElse(null);
+            resolvedId = route != null ? route.id : 0L;
+        }
+        else
+        {
+            RouteConfig route = binding.routes.stream()
+                .filter(r -> r.when.isEmpty())
+                .findFirst()
+                .orElse(null);
+            resolvedId = route != null ? route.id : 0L;
+        }
+
+        if (resolvedId != 0L)
```

Comment:
I don't think the guard for credentials is related to the need to check authorized on the routes, so only the `guard.reauthorize(...)` call is needed in the `if (guard != null)` block.

Recommend creating a `TestBindingConfig` and attaching that instead of generic `BindingConfig`, then have a `resolve` method on `TestBindingConfig` (used here) that always checks `route.authorized(...)` and `when.isEmpty()`.

That should work for all scenarios and also avoid the code repetition above.